test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp

```diff
@@ -698,6 +698,49 @@ TEST_F(StatisticsDomainParticipantTests, EnableStatisticsDataWriterFailureIncomp
 #endif // FASTDDS_STATISTICS
 }
 
+/**
+
+ * This test checks that a participant is correctly deleted by the factory
+ * after calling delete_contained_entities() method
+ * 1. Create a statistics participant, register type
+ * 2. Create a sample topic
+ * 3. Perform a delete_contained_entities() in the statistics participant
+ * 4. Delete the participant
+ */
+TEST_F(StatisticsDomainParticipantTests, DeleteParticipantAfterDeleteContainedEntitiesFailure)
```

Comment:
This test seems fine, but testing it without the fix does not give the expected result.
Expected result: it fails with a SegFault as the publisher of statistics have already been destroyed.
Actual result: it fails due to timeout as `delete_participant` get locked getting mutex `eprosima::fastdds::dds::DomainParticipantImpl::mtx_gs_`

Reply:
You are right, if you launch the test as it is, it gets blocked on that mutex. 
The thing is that the program in under undefined behaviour at this point (in fact, it is since the previous condition is skipped and enter the statement) because builtin_publisher_ is a dangling pointer.
```
if (nullptr != builtin_publisher_)
{... /*undefined behaviour*/...}
```
Try to launch the test with valgrind, the output will be the expected. We will discuss the best way to tackle this test.